src/pcre2/_cy.pyx

```diff
@@ -153,29 +163,28 @@ cdef inline void raise_from_rc(int rc):
 # ============================================================================
 #                                                          Pattern Compilation
 
-class CompileOption(IntEnum):
-    IGNORECASE = PCRE2_CASELESS
-    MULTILINE = PCRE2_MULTILINE
+
+class CompileOption(IntFlag):
+    CASELESS = PCRE2_CASELESS
     DOTALL = PCRE2_DOTALL
-    '''
-    Controls the input codec (whether the input bytes are read into characters by UTF-8
-    decoding). If the input pattern is a `str`, the default behaviour is UNICODE (and this
-    cannot be unset). If the input pattern is a `bytes`, the default is ASCII/Latin-1
-    (one byte per character), but UNICODE sets this to UTF-8.
-    '''
-    UNICODE = PCRE2_UTF
-    '''
-    Controls the interpretation of character values. If characters are ASCII, then
-    (for example) '\w' does not match values outside the range 0-127. If the input pattern
-    is a `str`, the default behaviour is UNICODE_PROPS (and this cannot be unset). If the
-    input pattern is a `bytes`, the default is ASCII, but UNICODE_PROPS sets this to
-    interpret character values according to Unicode.
-    '''
-    UNICODE_PROPS = PCRE2_UCP
-    VERBOSE = PCRE2_EXTENDED
-
-
-def compile(object pattern, uint32_t options = 0):
+    MULTILINE = PCRE2_MULTILINE
+    EXTENDED = PCRE2_EXTENDED
+
+    # Controls the input codec (whether the input bytes are read into characters by UTF-8
+    # decoding). If the input pattern is a `str`, the default behaviour is UNICODE (and this cannot
+    # be unset). If the input pattern is a `bytes`, the default is ASCII/Latin-1 (one byte per
+    # character), but UNICODE sets this to UTF-8.
+    UTF = PCRE2_UTF
+
+    # Controls the interpretation of character values. If characters are ASCII, then (for example)
+    # '\w' does not match values outside the range 0-127. If the input pattern is a `str`, the
+    # default behaviour is UNICODE_PROPS (and this cannot be unset). If the input pattern is a
```

Comment:
Comment needs updating. (It still refers to old name "UNICODE_PROPS"). It's also wrong when it says "this cannot be unset", now that you have the disabled_options improvement.